### ISSUE TITLE
VSCode unable to retrieve typing information with composite, emitDeclarationOnly and strict

### ISSUE THREAD

**Issue by JasonGore** [microsoft/TypeScript#45665]
# TL;DR

Exactly the same as #43909 with addition of `"strict": true` re-exposing issue, and also tested and reproing against TS 4.3.5 and 4.4.2. I've updated the same repro to cover this case.

# Bug Report

### 🔎 Search Terms

composite project references emit declaration emitdeclarationonly vscode

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about: declarationEmitOnly and composite/project references

I've tried:
* 4.2.4
* 4.3.0-beta
* 4.3.0-dev.20210430
* VSCode: 1.55.2
* 4.3.5
* 4.4.2

### ⏯ Playground Link

I don't think it's possible to set up multiple projects using project references in Playground. If I'm mistaken, please let me know. For now I've created the smallest repro I can below.

### 💻 Code

As this uses project references I set up [a basic monorepo](https://github.com/JasonGore/ts-composite-declaration) with no external deps other than typescript.

To repro (also in README.md):
1. clone https://github.com/JasonGore/ts-composite-declaration.git
1. run `yarn`
1. `cd packages/emit-standard`
1. `yarn tsc`
1. `cd ../packages/emit-composite`
1. `yarn tsc` (although I don't think this should technically be needed for projrefs)

Open VSCode from repo root and open `packages/consumer/src/index.ts` and note comments.

### 🙁 Actual behavior

This is an `emitDeclarationOnly` example emitting type information for packages that have `.js` files. The only difference between `emit-standard` and `emit-composite` is that one has `composite` set to true and is a reference of `consumer` package.

In VSCode, `emit-composite` seems unable to retrieve typing information, whether or not `emit-composite/lib` output exists:

![image](https://user-images.githubusercontent.com/26070760/116757828-b37d9200-a9c3-11eb-997f-44006a787fc5.png)

However, if you run `yarn tsc` in `packages/consumer`, typing information DOES seem to be available. I'm not sure if this is a TS or VSCode bug. For each TS version I tried above I ensured that VSCode was using "workspace version" and all exhibit this behavior.

### 🙂 Expected behavior

Typing information should be available in VSCode for both cases.
